contracts/HMToken.sol

```diff
@@ -12,47 +12,46 @@ contract HMToken is HMTokenInterface {
     using SafeMath for uint256;
     uint256 private constant MAX_UINT256 = 2**256 - 1;
     uint32  private constant BULK_MAX_COUNT = 100;
-    uint32  private constant BULK_MAX_VALUE = 10000000;
+    uint32  private constant BULK_MAX_VALUE;
 
     event BulkTransfer(uint256 indexed _txId, uint256 _bulkCount);
     event BulkApproval(uint256 indexed _txId, uint256 _bulkCount);
 
-    mapping (address => bool) public pauseTransfer;
-    mapping (address => uint256) public balances;
-    mapping (address => mapping (address => uint256)) public allowed;
+    mapping (address => uint256) private balances;
+    mapping (address => mapping (address => uint256)) private allowed;
 
     string public name;
     uint8 public decimals;
     string public symbol;
-    address public escrowFactory = 0;
 
     constructor(uint256 _totalSupply, string _name, uint8 _decimals, string _symbol) public {
-        totalSupply = _totalSupply;
+        totalSupply = _totalSupply * (10 ** uint256(_decimals));
         name = _name;
         decimals = _decimals;
         symbol = _symbol;
-        balances[msg.sender] = _totalSupply;
+        balances[msg.sender] = totalSupply;
+        BULK_MAX_VALUE = totalSupply;
```

Comment:
A constant is assignable on construction?

Reply:
Wow what a blunder, will fix that

Reply:
I honestly dont know, was just asking if its even possible